File: 2015/fsharp/Advent2015/Days/Day01.fs
module Days.Day01

open System.Text

let public solve () : unit =
    let input =
        System.IO.File.ReadAllText("../../inputs/day01.txt", Encoding.UTF8).Trim()

    let nums = [| for c in input -> if c = '(' then 1 else -1 |]
    let part2 =
        nums
        |> Seq.scan (+) 0
        |> Seq.indexed
        |> Seq.find (fun (_, s) -> s = -1)
        |> fst
    printfn $"\tPart 1: {Array.sum nums}"
    printfn $"\tPart 2: {part2}"

File: 2015/python/day01.py
from itertools import accumulate

def day01():
    input: str = open('../inputs/day01.txt', encoding='utf-8').read().strip()
    nums = [1 if c == '(' else -1 for c in input]
    print(f"Part 1: {sum(nums)}")
    print(f"Part 2: {next(i for i, s in enumerate(accumulate(nums)) if s == -1) + 1}")